File: codegen/results/v001/qwen3.5_4b_think/20260408_204208/python/008_prime_numbers/solution.py
import sys

def is_prime(num):
    """Check if a number is prime."""
    if num < 2:
        return False
    if num == 2:
        return True
    if num % 2 == 0:
        return False
    for i in range(3, int(num ** 0.5) + 1, 2):
        if num % i == 0:
            return False
    return True

def main():
    # Accept a single integer argument N from the command line
    if len(sys.argv) < 2:
        sys.exit(1)
        
    try:
        n = int(sys.argv[1])
    except ValueError:
        sys.exit(1)

    # Print all prime numbers up to and including N, one per line
    for i in range(2, n + 1):
        if is_prime(i):
            print(i)

if __name__ == "__main__":
    main()

File: codegen/results/v001/qwen3.5_4b_think/20260408_204208/typescript/008_prime_numbers/solution.ts
import { argv } from 'process';

const parseArgument = (args: string[]) => {
  for (let i = 2; i < args.length; i++) {
    const num = parseInt(args[i], 10);
    if (Number.isNaN(num)) {
      process.exit(1);
    }
    return num;
  }
  process.exit(1);
};

const sievePrimes = (n: number): number[] => {
  if (n < 2) {
    return [];
  }
  
  const isPrime = new Array<boolean>(n + 1).fill(true);
  isPrime[0] = false;
  isPrime[1] = false;
  
  for (let i = 2; i <= Math.floor(Math.sqrt(n)); i++) {
    if (isPrime[i]) {
      for (let j = i * i; j <= n; j += i) {
        isPrime[j] = false;
      }
    }
  }
  
  const primes: number[] = [];
  for (let i = 2; i <= n; i++) {
    if (isPrime[i]) {
      primes.push(i);
    }
  }
  
  return primes;
};

const printPrimes = (primes: number[]) => {
  for (const prime of primes) {
    console.log(prime);
  }
};

const main = () => {
  const N = parseArgument(argv);
  const primes = sievePrimes(N);
  printPrimes(primes);
};

main();